source/extensions/stat_sinks/open_telemetry/open_telemetry_impl.cc

```diff
@@ -18,32 +186,49 @@ generateResourceAttributes(const Tracers::OpenTelemetry::Resource& resource) {
   return resource_attributes;
 }
 
+Matcher::MatchTreePtr<Stats::StatMatchingData>
+createMatcher(const xds::type::matcher::v3::Matcher& matcher_config,
+              Server::Configuration::ServerFactoryContext& server_factory_context) {
+  ActionValidationVisitor validation_visitor;
+  ActionContext action_context;
+  Matcher::MatchTreeFactory<Stats::StatMatchingData, ActionContext> factory{
+      action_context, server_factory_context, validation_visitor};
+  return factory.create(matcher_config)();
+}
+
 OtlpOptions::OtlpOptions(const SinkConfig& sink_config,
-                         const Tracers::OpenTelemetry::Resource& resource)
+                         const Tracers::OpenTelemetry::Resource& resource,
+                         Server::Configuration::ServerFactoryContext& server)
     : report_counters_as_deltas_(sink_config.report_counters_as_deltas()),
       report_histograms_as_deltas_(sink_config.report_histograms_as_deltas()),
       emit_tags_as_attributes_(
           PROTOBUF_GET_WRAPPED_OR_DEFAULT(sink_config, emit_tags_as_attributes, true)),
       use_tag_extracted_name_(
           PROTOBUF_GET_WRAPPED_OR_DEFAULT(sink_config, use_tag_extracted_name, true)),
       stat_prefix_(!sink_config.prefix().empty() ? sink_config.prefix() + "." : ""),
-      resource_attributes_(generateResourceAttributes(resource)) {}
+      resource_attributes_(generateResourceAttributes(resource)),
+      matcher_(createMatcher(sink_config.custom_metric_conversions(), server)) {}
 
 OpenTelemetryGrpcMetricsExporterImpl::OpenTelemetryGrpcMetricsExporterImpl(
     const OtlpOptionsSharedPtr config, Grpc::RawAsyncClientSharedPtr raw_async_client)
     : config_(config), client_(raw_async_client),
       service_method_(*Protobuf::DescriptorPool::generated_pool()->FindMethodByName(
-          "opentelemetry.proto.collector.metrics.v1.MetricsService.Export")) {}
+          "opentelemetry.proto.collector.metrics.v1.MetricsService."
+          "Export")) {}
 
 void OpenTelemetryGrpcMetricsExporterImpl::send(MetricsExportRequestPtr&& export_request) {
+  ENVOY_LOG(debug, "sending the OTLP request \n{}", export_request->ShortDebugString());
```

Comment:
We should probably not log at debug level successful requests, since they are quite large.

Reply:
removed.